src/components/TableDialog.vue

```diff
@@ -95,109 +95,109 @@
 </template>
 
 <script lang="ts">
-import Vue, { PropType } from 'vue';
+import { defineComponent, ref, computed } from '@vue/composition-api';
 
 import api from '@/api';
 import { FileType } from '@/types';
 import { validFileType, fileName as getFileName } from '@/utils/files';
 
 const defaultKeyField = '_key';
-export default Vue.extend({
-  name: 'TableDialog',
+const filesTypes: FileType[] = [
+  {
+    extension: ['csv'],
+    queryCall: 'csv',
+    hint: 'Comma Separated Value file',
+    displayName: 'CSV',
+  },
+];
 
+export default defineComponent({
+  name: 'TableDialog',
   props: {
     workspace: {
-      type: String as PropType<string>,
+      type: String,
```

Comment:
Is there a difference between `string` and `String`? Why do we prefer the capitalized version here?

Reply:
`String` is a JavaScript type, while `string` is a TypeScript type.

The `props` declaration requires a JavaScript type to be given here, while the `as` bit allows that JavaScript type to be cast to a pure TypeScript type (using the `PropType` wrapper). This isn't specifically useful here (though, personally, I would probably just keep it), but it is in situations where a JavaScript type like `Object` provides much less information than a TypeScript type like `{ foo: string; bar: number; }`.

Reply:
`String` is the string class constructor, while `string` is the TypeScript string type. While it may look like we're typing the prop here (I mean semantically that _is_ what we're doing), this is an object, and so whatever is on the right side needs to be a value, specifically a constructor ([relevant](https://vuejs.org/v2/guide/components-props.html#Prop-Types) Vue docs), so that Vue can do it's thing at runtime. It didn't seem like TypeScript needed me to specifically cast this as `PropType<string>` anymore, so I removed that part of it.

Reply:
Whoops looks like we commented at the same time, but I think we both answered your question.

Reply:
Thanks for the explanations

Reply:
@AlmightyYakob's explanation is more accurate than mine, since JavaScript doesn't *really* have types. Mine was more of a gloss over what's really happening.